spec/unit/auto_link_spec.rb

```diff
@@ -1,60 +1,68 @@
 require 'rails_helper'
 
-RSpec.describe "auto linking resources" do
+RSpec.describe "auto linking resources", type: :view do
   include ActiveAdmin::ViewHelpers::ActiveAdminApplicationHelper
   include ActiveAdmin::ViewHelpers::AutoLinkHelper
   include ActiveAdmin::ViewHelpers::DisplayHelper
   include MethodOrProcHelper
 
   let(:active_admin_config)   { double namespace: namespace }
-  let(:active_admin_namespace){ ActiveAdmin::Namespace.new(ActiveAdmin::Application.new, :admin) }
+  let(:active_admin_namespace){ ActiveAdmin.application.namespace(:admin) }
   let(:post){ Post.create! title: "Hello World" }
 
-  def authorized?(*)
-    true
+  before do
+    allow(self).to receive(:authorized?).and_return(true)
   end
 
   context "when the resource is not registered" do
+    before do
+      load_resources {}
+    end
+
     it "should return the display name of the object" do
       expect(auto_link(post)).to eq "Hello World"
     end
   end
 
   context "when the resource is registered" do
     before do
-      active_admin_namespace.register Post
+      load_resources do
+        active_admin_namespace.register Post
+      end
     end
+
     it "should return a link with the display name of the object" do
-      url_path = "/admin/posts/#{post.id}?locale=en"
-      expect(self).to receive(:url_options).and_return(locale: 'en')
-      expect(self).to receive(:url_for) { |url| url }
-      expect(self).to receive(:link_to).with "Hello World", url_path
-      auto_link(post)
+      expect(auto_link(post)).to \
+          match(%r{<a href="/admin/posts/\d+">Hello World</a>})
     end
 
     context "but the user doesn't have access" do
+      before do
+        allow(self).to receive(:authorized?).and_return(false)
+      end
+
       it "should return the display name of the object" do
-        expect(self).to receive(:authorized?).twice.and_return(false)
         expect(auto_link(post)).to eq "Hello World"
       end
     end
+  end
 
-    context "but the show action is disabled" do
-      before do
+  context "when the resource is registered with the show action disabled" do
+    before do
+      load_resources do
         active_admin_namespace.register(Post) { actions :all, except: :show }
       end
+    end
 
-      it "should fallback to edit" do
-        url_path = "/admin/posts/#{post.id}/edit?locale=en"
-        expect(self).to receive(:url_options).and_return(locale: 'en')
-        expect(self).to receive(:url_for) { |url| url }
-        expect(self).to receive(:link_to).with "Hello World", url_path
-        auto_link(post)
-      end
+    it "should fallback to edit" do
+      expect(auto_link(post)).to \
+        match(%r{<a href="/admin/posts/\d+/edit">Hello World</a>})
```

Comment:
Right, I accidentally removed that while refactoring. The test description was not related to locales, so I forgot to re-add it. I think those tests should be separate from the "normal case" because otherwise we don't test the "normal case". So I added be2d3004861bc12d9c682cb7ce2a359d44dbfe72, what do you think?